include/seastar/core/seastar.hh

```diff
@@ -336,6 +336,18 @@ future<> sync_directory(std::string_view name) noexcept;
 /// containing directory is sync'ed.
 future<> remove_file(std::string_view name) noexcept;
 
+/// Removes a regular file via blocks discarding.
+///
+/// \param name name of the file to remove.
+///
+/// \note
+/// The removal is processed by the I/O queue according to its configuration.
+/// The removal may be delayed when the bandwidth is not available. The user
+/// must ensure, that the file is not used until the removal finishes.
+///
+/// Effectively: opens a file, unlinks it and punches holes until nothing is left.
+future<> remove_file_via_blocks_discarding(std::string_view name) noexcept;
```

Comment:
seastar.hh is too generic. The include/seastar/util/file.hh seems to be better place